fastify.js

```diff
@@ -121,21 +123,34 @@ function fastify (options) {
   const { logger, hasLogger } = createLogger(options)
 
   // Update the options with the fixed values
-  options.connectionTimeout = options.connectionTimeout || defaultInitOptions.connectionTimeout
-  options.keepAliveTimeout = options.keepAliveTimeout || defaultInitOptions.keepAliveTimeout
+  options.timeouts.connectionTimeout = options.timeouts.connectionTimeout || defaultInitOptions.connectionTimeout
+  options.timeouts.keepAliveTimeout = options.timeouts.keepAliveTimeout || defaultInitOptions.keepAliveTimeout
   options.logger = logger
   options.genReqId = genReqId
   options.requestIdHeader = requestIdHeader
   options.querystringParser = querystringParser
   options.requestIdLogLabel = requestIdLogLabel
   options.disableRequestLogging = disableRequestLogging
   options.ajv = ajvOptions
-  options.clientErrorHandler = options.clientErrorHandler || defaultClientErrorHandler
-  options.exposeHeadRoutes = exposeHeadRoutes
-
-  const initialConfig = getSecuredInitialConfig(options)
+  options.errorManagement.clientErrorHandler = options.errorManagement.clientErrorHandler || defaultClientErrorHandler
+  options.routing.exposeHeadRoutes = exposeHeadRoutes
+
+  const {
+    errorManagement,
+    routing,
+    security,
+    timeouts,
+    ...rest
+  } = options
+  const initialConfig = getSecuredInitialConfig({
+    ...errorManagement,
+    ...routing,
+    ...security,
+    ...timeouts,
+    ...rest
+  })
```

Comment:
Please don't do this. We should use the new format for this not the old one.

Reply:
@mcollina, this would also imply a corresponding change in schema for validation in ajv right?

Reply:
Yes exactly.

Essentially:

1. transform old options to new ones, adding warnings for each one of them.
2. validate
3. use new options everywhere.

Reply:
Gotcha, right on it